### ISSUE TITLE
ansible collection for the client side of things

### ISSUE THREAD

**Issue by bendem** [ansible-community/ara#461]
## What is the idea ?

Following the spirit of #71, it would be nice to install the callback and ansible modules using collections rather than through pip.

Currently, I have this is my ansible config to make the callback plugin useable, would love to clean it up: 

```ini
[defaults]
; non portable, only works when launched from our bundled container
callback_plugins = /venv/lib/python3.8/site-packages/ara/plugins/callback
```

I'm currently using `1.5.8`.

**Comment by dmsimard** [comment]
Hi @bendem and thanks for the issue.

I agree that it's challenging to keep track of where the callback is installed depending on the distro, venv, etc.
It's why we came up with helper modules like ``python3 -m ara.setup.callback_plugins`` but it's not something you can put in an ansible.cfg because it's just an ini file.

We dicussed whether we should do this a while back here:
- https://github.com/ansible-community/ara-collection/issues/9
- https://github.com/ansible-community/ara-collection/pull/30

At a high level, we've decided to keep the ansible plugins in the ara package because installing it takes care of the dependencies and lets us keep the simple "getting started" workflow:
- ``pip install ara[server]``
- ``export ANSIBLE_CALLBACK_PLUGINS=$(python3 -m ara.setup.callback_plugins)``
- Running a playbook as usual

If the callback is in the collection, there are several gotchas and additional steps required.

ara-collection has historically been about setting up an ara server but there could probably be a role that takes care of setting up the client side as well. That doesn't mean moving the callback to the collection, though.

Let me know if you have ideas but in the meantime I hope that makes sense.

**Comment by amilcarlucas** [comment]
ansible-lint complains that "ara_record" is not a FQCN .
is there a way to use a real FQCN without installing ara via collections?

**Comment by dmsimard** [comment]
@amilcarlucas that's a good question, I'm not sure.

While I realize it isn't ideal, I suppose a workaround would be to configure ansible-lint ignore that particular lint rule for the time being.

**Comment by bendem** [comment]
Yes, that's what we've done for now, but as you said, it's not ideal since it's a global setting and hides other valid problems.

**Comment by flowerysong** [comment]
The FQCN is `ansible.legacy.ara_record`, but it's a pretty useless linting rule so there's not really any danger to disabling it.

**Comment by dmsimard** [comment]
Oh, thanks @flowerysong, indeed that's what I was told when I asked about it in the ansible-lint channel and I forgot to follow up here.

**Comment by hille721** [comment]
> ansible-lint complains that "ara_record" is not a FQCN .

In my case ansible-lint is not complaining about FQCN but printing a warning when using `ara_playbook` 

```
WARNING  Unable to load module ara_playbook at playbooks/tests/test_ara_demo.yml:9 for options validation
WARNING  Unable to resolve FQCN for module ara_playbook
```

Wouldn't it be easy to write a proper Ansible collection for the ara plugins which simple wraps around the PyPI package?

**Comment by hille721** [comment]
Also no syntax highlighting, autocomplete or documentation on hover for the ara modules when using Ansible extension in VSCode

**Comment by bendem** [comment]
> In my case ansible-lint is not complaining about FQCN but printing a warning when using `ara_playbook`

Did you try specifying `ansible.legacy.ara_playbook` as suggested?

**Comment by hille721** [comment]
> Did you try specifying `ansible.legacy.ara_playbook` as suggested?

As I said, I don't have a problem regarding the FQCN, I just get those warnings. If I use `ansible.legacy.ara_playbook` it changes nothing except the warning is now

```
WARNING  Unable to load module ansible.legacy.ara_playbook at tests/test_ara_demo.yml:9 for options validation
WARNING  Unable to resolve FQCN for module ansible.legacy.ara_playbook

**Comment by bendem** [comment]
arf, fun times.

**Comment by Hipska** [comment]
As a new user to ARA, I was also confused about the way to install and use the callback. IMHO having it as a collection (which wraps the PyPi package as @hille721 suggested) would make much sense.

**Comment by hille721** [comment]
@Hipska in fact I already discussed this with @dmsimard and @Thulium-Drake last year and already opened a WIP PR: https://github.com/ansible-community/ara-collection/pull/75
Unfortunately I did not had the time to further drive it to get it merged.
But basically it works, already using that approach for a while now.
